src/transfer.cpp

```diff
@@ -387,6 +387,7 @@ void Transfer::failed(error e, DBTableTransactionCommitter& committer, dstime ti
 
     if (e == API_EOVERQUOTA)
     {
+        assert(type != PUT || !timeleft); //only expected overstorage overquota for uploads
```

Comment:
I think we can check both PUT and GET cases here - some code relies on testing only timeleft, so these two have to match:    `assert((type == PUT && !timeleft) || (type == GET && timeleft)); `